app/ui/src/Components.elm

```diff
@@ -0,0 +1,33 @@
+module Components exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput, onSubmit)
+import Model exposing (Output(..))
+
+
+layout : List (Html msg) -> Html msg
+layout children =
+    div [] children
+
+
+query : (String -> msg) -> msg -> Html msg
+query queryChange submit =
+    Html.form [ onSubmit submit ]
+        [ label [ for "query" ] [ text "Your query: " ]
+        , input [ id "query", onInput queryChange ] []
+        , button [ type_ "submit", onClick submit ] [ text "Update" ]
+        ]
+
+
+output : Output -> Html msg
```

Comment:
This is very true. This is 

1. Me playing with Union Types trying to understand how one works with them
2. Preparing for Rendering a status message OR a HAR view, which doesn't contain a simple message as a thing.